### PR TITLE
Backend/feat/added-the-support-of-registry-map-fallback-table

### PR DESCRIPTION
**Pull request by prashant601**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
shared-kernel-changes: https://github.com/nammayatri/shared-kernel/pull/236

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [x] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/RegistryMapFallback.hs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+module Domain.Types.RegistryMapFallback where
+
+import Domain.Types.Merchant
+import Kernel.Prelude
+import Kernel.Types.Id
+
+data RegistryMapFallback = RegistryMapFallback
+  { subscriberId :: Text,
+    uniqueId :: Text,
+    merchantId :: Id Merchant,
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/RegistryMapFallback.hs:24]: in case of gateway there won't be any merchantID, so make this Maybe.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+cacheRegistryMapFallback :: CacheFlow m r => Text -> Maybe RegistryMapFallback -> m ()
+cacheRegistryMapFallback key registryMap = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  Hedis.setExp key registryMap expTime
+
+makeSubscriberIdKey :: Text -> Text
+makeSubscriberIdKey subscriberId = "driver-offer:CachedQueries:RegistryMapFallback:SubscriberId-" <> subscriberId
+
+makeUniqueIdKey :: Text -> Text
+makeUniqueIdKey uniqueId = "driver-offer:CachedQueries:RegistryMapFallback:uniqueId-" <> uniqueId
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs:63]: Make U capital, to follow the semantics.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+makeMerchantIdKey :: Id Merchant -> Text
+makeMerchantIdKey mId = "driver-offer:CachedQueries:RegistryMapFallback:MerchantId-" <> mId.getId
+
+makeSubscriberIdAndUniqueIdKey :: Text -> Text -> Text
+makeSubscriberIdAndUniqueIdKey subscriberId uniqueId = "driver-offer:CachedQueries:RegistryMapFallback:SubscriberId-" <> subscriberId <> "UniqeId-" <> uniqueId
```
Comment (RitikaHanish) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs:69]: Unique typo

**Review by RitikaHanish — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs (hunk 3 above)
Comment (prashant601) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs]: modified

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs (hunk 2 above)
Comment (prashant601) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/RegistryMapFallback.hs]: done

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/RegistryMapFallback.hs (hunk 1 above)
Comment (prashant601) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/RegistryMapFallback.hs]: as discussed, I've completely removed the merchant id

File: Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall [hunk 4]
```diff
@@ -26,15 +26,7 @@ let esqDBReplicaCfg =
 
 let esqLocationDBCfg = esqDBCfg
```
Comment (RitikaHanish) [inline Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall:28]: Revert this change.

**Review by RitikaHanish — CHANGES_REQUESTED** [review]
Add drainer changes

File: Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall (hunk 4 above)
Comment (prashant601) [reply Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall]: done

**Review by RitikaHanish — APPROVED** [review]
